src/main/java/de/ait/secondlife/controllers/OfferController.java

```diff
@@ -0,0 +1,211 @@
+package de.ait.secondlife.controllers;
+
+
+import de.ait.secondlife.domain.dto.ResponseMessageDto;
+import de.ait.secondlife.domain.dto.OfferCreationDto;
+import de.ait.secondlife.domain.dto.OfferResponseDto;
+import de.ait.secondlife.domain.dto.OfferResponseWithPaginationDto;
+import de.ait.secondlife.domain.dto.OfferUpdateDto;
+import de.ait.secondlife.exception_handling.OfferExceptionHandler;
+import de.ait.secondlife.exception_handling.exeptions.PaginationParameterIsWrongException;
+import de.ait.secondlife.services.interfaces.OfferService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/v1/offer")
```

Comment:
in RestApi it is better to use the plural form - /v1/offer**s**

Reply:
The approach of naming the base path in the name of an entity has been used. I'll fix it.